views/js/previewer/plugins/content/enhancedReadOnlyMode.js

```diff
@@ -0,0 +1,90 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Test Previewer Content Plugin : EnhancedReadOnlyMode
+ *
+ * This plugin can be used as a hook to do modification in the item preview for read only mode
+ *
+ * @author Ansul Sharma <ansul@taotesting.com>
+ */
+define([
+    'jquery',
+    'lodash',
+    'i18n',
+    'taoTests/runner/plugin'
+], function ($, _, __, pluginFactory) {
+    'use strict';
+
+    return pluginFactory({
+
+        name: 'EnhancedReadOnlyMode',
+
+        /**
+         * Initialize the plugin (called during runner's init)
+         */
+        init() {
+            const testRunner = this.getTestRunner();
+
+            /**
+             * Enables the plugin only in readOnly mode
+             * @returns {Boolean}
+             */
+            const isPluginAllowed = () => {
+                const config = testRunner.getConfig();
+                return config.options.readOnly;
+            };
+
+            testRunner
+                .after('renderitem', () => {
+                    if (isPluginAllowed()) {
+                        const $contentArea = testRunner.getAreaBroker().getContentArea();
+                        const $extendedTextinteractionTextAreas = $contentArea.find('.qti-extendedTextInteraction textarea.text-container');
+                        const $ckeEditorsContent = $contentArea.find('.qti-extendedTextInteraction div.cke_contents');
+
+                        /**
+                         * Updates the height of textarea element of all extended text interactions based on the height of the content
+                         */
+                        if($extendedTextinteractionTextAreas.length) {
+                            $extendedTextinteractionTextAreas.map((i, $textArea) => {
+                                $textArea.style.height = `${$textArea.scrollHeight + 20}px`;
+                            });
+                        }
+
+                        /**
+                         * Updates the height of all the ckeEditor container of wysiwyg extended text interaction based on the height of the iFrame
+                         */
+                        if($ckeEditorsContent.length) {
+                            $ckeEditorsContent.map((i, ckeEditorContent) => {
+                                const $ckeEditorContent = $(ckeEditorContent);
+                                const $ckeEditorIFrame = $ckeEditorContent.find('iframe.cke_wysiwyg_frame');
+
+                                /**
+                                 * Only update the height when the iFrame has finished loading the styles because font-size may change the height
+                                 */
+                                $ckeEditorIFrame.load = setTimeout(() => {
```

Comment:
How can this work? If this is a jQuery collection, you are redefining the load method...

Reply:
It would still work because I think jquery executes this fn whenever `load` is triggered but it's definitely not the right way to do it. I have changed it to `.on('load', () => {})`